src/js/views/globals/add-patient/add-patient_views.js

```diff
@@ -0,0 +1,298 @@
+    <button class="button--text u-margin--r-4 js-cancel">{{ @intl.globals.addPatient.addPatientViews.saveView.cancelBtn }}</button>
+  `,
+  triggers: {
+    'click .js-cancel': 'cancel',
+    'click .js-save': 'save',
+  },
+  templateContext() {
+    return {
+      isDisabled: !this.model.isValid() || this.isDisabled,
+    };
+  },
+  onSave() {
+    this.isDisabled = true;
+    this.render();
+  },
+});
+
+const SexDroplist = Droplist.extend({
+  initialize() {
+    this.collection = new Backbone.Collection([
+      {
+        text: 'Male',
+        value: 'm',
+      },
+      {
+        text: 'Female',
+        value: 'f',
+      },
+      {
+        text: 'Other',
+        value: 'other',
+      },
+    ]);
+  },
+  viewOptions: {
+    className: 'button-secondary',
+    template: hbs`{{far "user"}}{{ text }}{{#unless text}}{{ @intl.globals.addPatient.addPatientViews.sexDroplist.defaultText }}{{/unless}}`,
+  },
+});
+
+const BirthdateView = View.extend({
+  tagName: 'button',
+  template: hbs`
+    <button class="button-secondary js-date {{#if hasError}}has-error{{/if}}">
+      {{far "calendar-alt"}}{{formatDateTime birth_date "LONG" defaultHtml=(intlGet "globals.addPatient.addPatientViews.birthDateView.defaultText")}}
+    </button>
+  `,
+  templateContext() {
+    const state = this.getOption('state');
+
+    return {
+      hasError: state.get('errorField') === 'birth_date',
+    };
+  },
+  ui: {
+    'date': '.js-date',
+  },
+  triggers: {
+    'click @ui.date': 'click',
+  },
+  modelEvents: {
+    'change:birth_date': 'render',
+  },
+  initialize() {
+    this.listenTo(this.getOption('state'), {
+      'change:errorField': this.render,
+    });
+  },
+  onClick() {
+    const datepicker = new Datepicker({
+      ui: this.ui.date,
+      uiView: this,
+      state: {
+        selectedDate: this.model.get('birth_date'),
+      },
+    });
+
+    this.listenTo(datepicker, {
+      'change:selectedDate'(date) {
+        this.model.set('birth_date', date.format('YYYY-MM-DD'));
+        datepicker.destroy();
+      },
+    });
+
+    datepicker.show();
+  },
+});
+
+const FirstNameView = View.extend({
+  template: hbs`<input class="input-primary w-100 js-input {{#if hasError}}has-error{{/if}}" value="{{ first_name }}" />`,
+  templateContext() {
+    const state = this.getOption('state');
+
+    return {
+      hasError: state.get('errorField') === 'name',
+    };
+  },
+  behaviors: [InputWatcherBehavior],
+  ui: {
+    input: '.js-input',
+  },
+  initialize() {
+    this.listenTo(this.getOption('state'), {
+      'change:errorField': this.render,
+    });
+  },
+  onDomRefresh() {
+    this.ui.input.focus();
+  },
+  onWatchKeydown(evt) {
+    if (evt.which === ENTER_KEY) {
+      evt.preventDefault();
+      return;
+    }
+  },
+  onWatchChange(text) {
+    const newText = removeNewline(text);
```

Comment:
is newline an issue here?  I kinda thing an `input` won't matter and that this is needed for textareas or contenteditables